docs/src/man/basics.md

```diff
@@ -0,0 +1,2042 @@
+   4 │     3     45  male        2  free     little           little
+   5 │     4     53  male        2  free     little           little           ⋯
+                                                               3 columns omitted
+
+julia> german[[1, 6, 15], :]
+3×10 DataFrame
+ Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking account ⋯
+     │ Int64  Int64  String  Int64  String   String           String           ⋯
+─────┼──────────────────────────────────────────────────────────────────────────
+   1 │     0     67  male        2  own      NA               little           ⋯
+   2 │     5     35  male        1  free     NA               NA
+   3 │    14     28  female      2  rent     little           little
+                                                               3 columns omitted
+
+julia> german[:, [:Age, :Sex]]
+1000×2 DataFrame
+  Row │ Age    Sex
+      │ Int64  String
+──────┼───────────────
+    1 │    67  male
+    2 │    22  female
+    3 │    49  male
+    4 │    45  male
+    5 │    53  male
+    6 │    35  male
+    7 │    53  male
+    8 │    35  male
+  ⋮   │   ⋮      ⋮
+  994 │    30  male
+  995 │    50  male
+  996 │    31  female
+  997 │    40  male
+  998 │    38  male
+  999 │    23  male
+ 1000 │    27  male
+      985 rows omitted
+```
+
+In this example we have created a `DataFrame` using `:Sex` and `:Age` column for the first five Rows:
+
+```jldoctest dataframe
+julia> german[1:5, [:Sex, :Age]]
+5×2 DataFrame
+ Row │ Sex     Age
+     │ String  Int64
+─────┼───────────────
+   1 │ male       67
+   2 │ female     22
+   3 │ male       49
+   4 │ male       45
+   5 │ male       53
+```
+
+Pay attention that `german[!, [:Sex]]` and `german[:, [:Sex]]` return a data frame object,
+while `german[!, :Sex]` and `german[:, :Sex]` return a vector. In the first case, `[:Sex]` 
+is a vector, indicating that the resulting object should be a data frame. On the other hand,
+`:Sex` is a single symbol, indicating that a single column vector should be extracted. Note 
+that in the first case a vector is required to be passed (not just any iterable), so e.g. 
+`german[:, (:Age, :Sex)]` is not allowed, but `german[:, [:Age, :Sex]]` is valid.
+
+```jldoctest dataframe
+julia> german[!, [:Sex]]
+1000×1 DataFrame
+  Row │ Sex
+      │ String
+──────┼────────
+    1 │ male
+    2 │ female
+    3 │ male
+    4 │ male
+    5 │ male
+    6 │ male
+    7 │ male
+    8 │ male
+  ⋮   │   ⋮
+  994 │ male
+  995 │ male
+  996 │ female
+  997 │ male
+  998 │ male
+  999 │ male
+ 1000 │ male
+985 rows omitted
+
+julia> german[!, [:Sex]] == german[:, [:Sex]]
+true
+
+julia> german[!, :Sex]
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+
+julia> german[!, :Sex] == german[:, :Sex]
+true
+```
+
```

Comment:
```suggestion

As it was explained above the difference between using `!` and `:`
when passing a row index is that `!` does not perform a copy of columns, while `:` does.
Therefore `german[!, [:Sex]]` data frame stores the same vector as the source `german` data frame,
while `german[:, [:Sex]]` stores its copy. The `!` selector normally should be avoided
as using it can lead to hard to catch bugs. However, when working with very large data frames
it can be useful to save memory and improve performance of operations.
```